nengo/builder/connection.py

```diff
@@ -169,7 +171,7 @@ def get_prepost_signal(is_pre):
     # Sample transform if given a distribution
     transform = (conn.transform.sample(conn.size_out, conn.size_mid, rng=rng)
                  if isinstance(conn.transform, Distribution) else
-                 conn.transform)
+                 np.array(conn.transform))
```

Comment:
Sorry commented on your first post before reading the second.  Yeah, I think it's easiest/clearest to just do the copy here.  But that'd be a nice wishlist item, a config setting that'd act like a compiler flag to optimize for memory usage.